src/pal/pal.h

```diff
@@ -168,10 +168,23 @@ namespace snmalloc
     Pal::error(msg.get_message());
   }
 
+  static inline size_t get_tid()
+  {
+    static thread_local size_t tid{0};
+    static std::atomic<size_t> tid_source{1};
+
+    if (tid == 0)
+    {
+      tid = tid_source++;
+    }
+    return tid;
+  }
+
```

Comment:
Generally approve of the idea, but should this actually be in the PAL so we can actually get the platform notion of a thread ID (when it's sufficiently inexpensive to do so)?

Reply:
This is a strict improvement, so no reason to drop it.  Was mostly asking in case there was reason not to do the PAL thing.

Reply:
I was trying to trace a bug in `DecayRange` which only exhibited with concurrency, and this was the quickest thing I could do to get a trace I could understand.  I think we can engineer something better.

Reply:
> I think we can engineer something better.

But not tonight ;-)